lib/router/transition.js

```diff
@@ -58,7 +70,7 @@ function Transition(router, intent, state, error) {
       this.pivotHandler = handlerInfo.handler;
     }
 
-    this.sequence = Transition.currentSequence++;
+    this.sequence = router.currentSequence++;
     this.promise = state.resolve(checkForAbort, this)['catch'](function(result) {
```

Comment:
I'm still concerned about this accidentally trolling us in the future. One of us won't be paying attention and will merge a PR which accidentally again captures things in a closure. In general I'm hesitant to pass around "huge" (by reference) objects.